src/bundler.ts

```diff
@@ -30,32 +30,35 @@ import {FSUrlLoader} from 'polymer-analyzer/lib/url-loader/fs-url-loader';
 import constants from './constants';
 import * as matchers from './matchers';
 import PathResolver from './pathresolver';
-import ASTUtils from './ast-utils';
-
-
-function buildLoader(config: any) {
-  const abspath: string = config.abspath;
-  const excludes = config.excludes;
-  const fsResolver = config.fsResolver;
-  const redirects = config.redirects;
-  let root = abspath && path.resolve(abspath) || process.cwd();
-  let loader = new FSUrlLoader(root);
-  // TODO(garlicnation): Add noopResolver for external urls.
-  // TODO(garlicnation): Add redirectResolver for fakeprotocol:// urls
-  // TODO(garlicnation): Add noopResolver for excluded urls.
-  return loader;
+import * as ASTUtils from './ast-utils';
+
+
+export interface Options {
+  abspath?: string;
+  addedImports?: string[];
+  analyzer?: Analyzer;
+  excludes?: string[];
+  implicitStrip?: boolean;
+  inlineCss?: boolean;
+  inlineScripts?: boolean;
+  inputUrl?: string;
+  redirects?: string[];
+  stripComments?: boolean;
+  stripExcludes?: string[];
 }
 
+
 class Bundler {
-  constructor(opts: any) {
+  constructor(opts: Options) {
+    this.analyzer = opts.analyzer!;
     // implicitStrip should be true by default
     this.implicitStrip =
         opts.implicitStrip === undefined ? true : Boolean(opts.implicitStrip);
     this.abspath = (String(opts.abspath) === opts.abspath &&
                     String(opts.abspath).trim() !== '') ?
         path.resolve(opts.abspath) :
-        null;
-    this.pathResolver = new PathResolver(this.abspath);
+        undefined;
```

Comment:
Is this how clang is formatting a ternary? I'd rather see the `?` and ':' at the beginning of lines. Also, the `String()` boxing calls seem weird to me, isn't this just checking if the string is null or empty?

```
    this.abspath = (opts.abspath == null || opts.abspath.trim() == '')
      ? undefined
      : path.resolve(opts.abspath);
```

Also, also, `path.resolve(p)` implicitly depends on the CWD. I don't even know what abspath does, but if we're receiving a configured `Analyzer` is it necessary?